test/operator/comparison.yml

```
@@ -42,6 +41,17 @@ variables:
   - "[]"
 
 tests:
+- name: "Invalid"
+  result: []
+  variables:
+    op: ['<', '<=', '>', '>=']
+    expr:
+    - 'a[].b'
+    - 'a[].b[]'
+  tests:
+  - query: '*[42 ~op~ ~expr~]'
+  - query: '*[~expr~ ~op~ 42]'
```

With a slight tweak we can leverage the built-in generators and it should also generate join queries.

```suggestion
- name: "Invalid"
  result: null
  variables:
    op: ['<', '<=', '>', '>=']
    expr:
    - 'a[].b'
    - 'a[].b[]'
    num: 42
  tests:
  - query: '~num~ ~op~ ~expr~'
  - query: '~expr~ ~op~ ~num~'
```